Issue when Building Native library on Mac OS X with JDK 1.8

When trying to do a 

sh ./bootstrap

I get the following error:

./bootstrap: line 3: autoreconf: command not found

Here is the next hurdle:

checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... no
checking whether make supports nested variables... no
checking for gcc... gcc
checking whether the C compiler works... no
configure: error: in `/Users/joe/Downloads/flow-native-srcrelease':
configure: error: C compiler cannot create executables
See `config.log' for more details

The config.log shows the following... Looks like I need gcc > version 5?

configure:2956: found /usr/local/bin/gcc
configure:2967: result: gcc
configure:3196: checking for C compiler version
configure:3205: gcc --version >&5
gcc (GCC) 4.9.2 20141029 (prerelease)
Copyright (C) 2014 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

I now did the following:

export JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_60.jdk/Contents/Home/include/

and then ran the ./configure && make which gave me the following error:

Making all in src
/bin/sh ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..   -I/Library/Java/JavaVirtualMachines/jdk1.8.0_60.jdk/Contents/Home/include  -g -O2 -MT flow_jni.lo -MD -MP -MF .deps/flow_jni.Tpo -c -o flow_jni.lo flow_jni.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I/Library/Java/JavaVirtualMachines/jdk1.8.0_60.jdk/Contents/Home/include -g -O2 -MT flow_jni.lo -MD -MP -MF .deps/flow_jni.Tpo -c flow_jni.c  -fno-common -DPIC -o .libs/flow_jni.o
In file included from com_github_jodersky_flow_internal_NativeSerial.h:2:0,
                 from flow_jni.c:2:
/Library/Java/JavaVirtualMachines/jdk1.8.0_60.jdk/Contents/Home/include/jni.h:45:20: fatal error: jni_md.h: No such file or directory
 #include "jni_md.h"
                    ^
compilation terminated.
make[2]: *** [flow_jni.lo] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2

--- a/Documentation/building.md
+++ b/Documentation/building.md
@@ -39,8 +39,9 @@ Several steps are involved in producing the native library:
 
 2. Compile
    
-   1. Check availability of dependencies: C compiler and JDK
-   2. Run `./configure && make`
+    1. Check availability of dependencies: C compiler and JDK (1.7 or above)
+    2. Run `./configure && make`.
+       *Note: should you encounter an error about a missing "jni.h" file, try setting the JAVA_HOME environment variable to point to base path of your JDK installation.*
 
 3. Install
 

--- a/flow-native/m4/ax_jni_include_dir.m4
+++ b/flow-native/m4/ax_jni_include_dir.m4
@@ -65,11 +65,8 @@ else
     _JTOPDIR=`echo "$_ACJNI_FOLLOWED" | sed -e 's://*:/:g' -e 's:/[[^/]]*$::'`
 fi
 
-case "$host_os" in
-        darwin*)        _JTOPDIR=`echo "$_JTOPDIR" | sed -e 's:/[[^/]]*$::'`
-                        _JINC="$_JTOPDIR/Headers";;
-        *)              _JINC="$_JTOPDIR/include";;
-esac
+_JINC="$_JTOPDIR/include"
+
 _AS_ECHO_LOG([_JTOPDIR=$_JTOPDIR])
 _AS_ECHO_LOG([_JINC=$_JINC])
 
@@ -93,6 +90,7 @@ osf*)           _JNI_INC_SUBDIRS="alpha";;
 solaris*)       _JNI_INC_SUBDIRS="solaris";;
 mingw*)     _JNI_INC_SUBDIRS="win32";;
 cygwin*)    _JNI_INC_SUBDIRS="win32";;
+darwin*)    _JNI_INC_SUBDIRS="darwin";;
 *)              _JNI_INC_SUBDIRS="genunix";;
 esac
 

